---
title: "Task"
author: Yanyu Liang
date: "`r format(Sys.time(), '%d %B, %Y')`"
---

# Data preparation

We need to incorporate new data (fetal brain related) to the model. It contains two sub tasks:

1. Annotate DeepSEA sequence with new labels. 
    - See pipeline memo [here](memo_preprocessing.html)
2. Extract new sequences of interest based on new data set. 
    - See pipeline memo [here](memo_extract.html)

# Train new models

There are three strategies:

0. Baseline classifier. [here](baseline.html)
    - ROC/PR curve [JASPAR logistic](https://rawgit.com/liangyy/deep_variant/code/pr_roc_curve/report/JASPAR_plot.html), [JASPAR logistic, reweighted](https://rawgit.com/liangyy/deep_variant/code/pr_roc_curve/report/JASPAR.reweight_plot.html)
    - Evalution using allelic imbalance site **TODO**
    - Compare performance with DeepSEA representation [here](https://rawgit.com/liangyy/deep_variant/code/performance_comparison/report/deep_brain.auc_comparison.html), [with reweight](https://rawgit.com/liangyy/deep_variant/code/performance_comparison/report/deep_brain_reweight.auc_comparison.html)
1. Use DeepSEA sequences and DeepSEA feature representation. 
    - See results and analysis [here](type1.html). 
    - See performance on new sequences [here](type1_newseq.html)
    - ROC/PR curve [Type1](http://htmlpreview.github.io/?https://github.com/liangyy/deep_variant/blob/code/pr_roc_curve/report/deepsea_vs_newseq_plot.html), [Type1, reweighted](https://rawgit.com/liangyy/deep_variant/code/pr_roc_curve/report/type1.reweight_plot.html)
    - Evalution using allelic imbalance site [here](type1_allelic_imbalance.html)
2. Use DeepSEA sequences but train a new model using the same architecture as DeepSEA. 
    - See results and analysis [here](type2.html). **ONGOING**
3. Include new sequences and train a new model using the same architecture. **TODO**

# Other analysis

There are some other directions and issues we need to explore and tackle. The analysis of them are listed here.

1. **Motif Analysis**:
    - Examine which filter or a set of filters that contribute most to a particular prediction task. See details [here](motif_analysis.html). **ONGOING**
2. **Training Strategy**
    - How to boost the model to specialize on a particular prediction task, say on certain tissue type? (change architecture? how to deal with overfitting?) **TODO**
3. **Performance Evaluation**
    -How to measure the performance on imbalanced data. For instance, how to interpret ROC AUC and PR AUC? See related discussion [here](auc.html)
4. **Application**
    - IBD GWAS SNPs. See [here](SNP_enrichment.html) Rmarkdown script is at [here](https://uchicago.box.com/s/zdntptc8udymivtyoapxpu365rpo3b18)
5. **Data Quality**: 
    - GC content. [DeepSEA test set](https://rawgit.com/liangyy/deep_variant/code/gc_content/report/deepsea.gc_content.html), [Sequences extracted from BED file](https://rawgit.com/liangyy/deep_variant/code/gc_content/report/extracted.gc_content.html)

# TODO list

* 4/9/17:
    1. ~~ROC/PR AUC interpretation~~ [link](auc.html)
    2. ~~Extract sequences from new data sets~~ [link](memo_extract.html)
* 4/10/17:
    3. New loss function? Weighted hinge loss? [link](loss_function.html) **TODO**
    4. Motif analysis of current model [link](motif_analysis.html) **ONGOING**
* 4/14/17:
    5. ~~Build up pipeline using snakemake.~~ See issue1 at [link](https://github.com/CreRecombinase/DeepVariantPrediction)
* 4/26/17:
    6. ~~Build baseline classifier~~ [link](baseline.html)
* 5/29/17:
    7. GC content bias in training and testing data
        - To overcome this, we can either use unbiased sequence set or reweight the performance evaluation function based on the GC content.
        - At first, do reweighting (testing - training - motif analysis) **ONGOING**
        - Construct a sequence set with matched GC content in positive and negative instances (use Homer?) **TODO**






