<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Yanyu Liang" />


<title>Task</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-4.5.0/css/font-awesome.min.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}

.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Deep Variant</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Tasks</a>
</li>
<li>
  <a href="analysis_list.html">Analysis</a>
</li>
<li>
  <a href="memo_list.html">Technical Memo</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/liangyy/deep_variant">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Task</h1>
<h4 class="author"><em>Yanyu Liang</em></h4>
<h4 class="date"><em>03 June, 2018</em></h4>

</div>


<div id="data-preparation" class="section level1">
<h1><span class="header-section-number">1</span> Data preparation</h1>
<p>We need to incorporate new data (fetal brain related) to the model. It contains two sub tasks:</p>
<ol style="list-style-type: decimal">
<li>Annotate DeepSEA sequence with new labels.
<ul>
<li>See pipeline memo <a href="memo_preprocessing.html">here</a> for E081, E082, Noonan, E129 (DeepSEA label)</li>
<li>See pipeline memo <a href="memo_preprocessing_ATACseq.html">here</a> for ATAC-seq data from Noonan’s lab</li>
<li>See pipeline memo <a href="memo_preprocessing_GSE70823.html">here</a> for d27-NSC, d41-neuron data from GSE70823</li>
<li>See pipeline memo <a href="memo_preprocessing_atac_seq_053018.html">here</a> for ATAC-seq data obtained from Yifan 5/30/18</li>
</ul></li>
<li>Extract new sequences of interest based on new data set.
<ul>
<li>See pipeline memo <a href="memo_extract.html">here</a> for E081, E082, Noonan, E129 (DeepSEA label)</li>
<li>See pipeline memo <a href="memo_extract_ATACseq.html">here</a> for ATAC-seq data from Noonan’s lab</li>
<li>See pipeline memo <a href="memo_extract_GSE70823.html">here</a> for d27-NSC, d41-neuron data from GSE70823</li>
<li>See pipeline memo <a href="memo_extract_ATACseq053018.html">here</a> for ATAC-seq data obtained from Yifan 5/30/18</li>
</ul></li>
<li>Define positive and negative sequences
<ul>
<li>See memo <a href="memo_posneg_fantom.html">here</a> for fantom fetal brain cage peak from Zhongshan (06/02/18)</li>
</ul></li>
</ol>
</div>
<div id="genomic-data" class="section level1">
<h1><span class="header-section-number">2</span> Genomic Data</h1>
<ol style="list-style-type: decimal">
<li>Stage1
<ul>
<li>Fetal brain DNase-seq from Roadmap (E081, E082)</li>
<li>H3K27ac mark from Noonan’s lab (Noonan)</li>
<li>One label from Deepsea model (E129)</li>
</ul></li>
<li>Stage2
<ul>
<li>ATAC-seq data from Noonan’s lab (hNSC-50, hNSC-P15-1, hNSC-P5-1, hNSC-P5-2)</li>
</ul></li>
<li>paperNSC
<ul>
<li>NSC data from this <a href="https://www.ncbi.nlm.nih.gov/pubmed/28803920">paper</a> and the data is obtained from this <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE70823">link</a></li>
<li>d27, d41-1, d41-2 data are used</li>
<li>Is referred as <code>paperNSC</code></li>
</ul></li>
<li>ATAC053018
<ul>
<li>Obtained from Yifan on 5/30/18</li>
<li>Contains cell types labelled as <code>CN</code>, <code>DN</code>, <code>GA</code>, <code>ips</code>, <code>NSC</code></li>
</ul></li>
</ol>
</div>
<div id="train-new-models" class="section level1">
<h1><span class="header-section-number">3</span> Train new models</h1>
<p>There are three strategies:</p>
<ol start="0" style="list-style-type: decimal">
<li>Baseline classifier. <a href="baseline.html">here</a>
<ul>
<li>Train an SVM or logistic classifier using JASPAR as annotation</li>
<li>Gkm-SVM classifier is trained on 5000 positive and 5000 negative sequences one annotation at a time.</li>
</ul></li>
<li>Use DeepSEA sequences and DeepSEA feature representation.
<ul>
<li>See training result <a href="type1.html">Stage1 here</a>.</li>
<li>See performance on new sequences <a href="type1_newseq.html">Stage1 here</a></li>
<li>Evalution using allelic imbalance site <a href="type1_allelic_imbalance.html">Stage1 here</a></li>
</ul></li>
<li>Use DeepSEA sequences but train a new model using the same architecture as DeepSEA.
<ul>
<li>See results and analysis <a href="type2.html">Stage1 here</a>. <strong>Leave for future</strong></li>
</ul></li>
<li>Include new sequences and train a new model using the same architecture. <strong>Leave for future</strong></li>
</ol>
</div>
<div id="results" class="section level1">
<h1><span class="header-section-number">4</span> Results</h1>
<ol style="list-style-type: decimal">
<li>ROC/PR curve
<ul>
<li><a href="https://rawgit.com/liangyy/deep_variant/code/pr_roc_curve/report/JASPAR_plot.html">Stage1, JASPAR logistic, deepsea+extracted, no reweight</a></li>
<li><a href="https://rawgit.com/liangyy/deep_variant/code/pr_roc_curve/report/JASPAR.reweight_plot.html">Stage1 JASPAR logistic, deepsea+extracted, reweight</a></li>
<li><a href="https://rawgit.com/liangyy/deep_variant/code/pr_roc_curve/report/type1.gkm_deepsea_plot.html">Stage1 gkmsvm, deepsea, no reweight</a></li>
<li><a href="https://rawgit.com/liangyy/deep_variant/code/pr_roc_curve/report/type1.gkm_deepsea.reweight_plot.html">Stage1 gkmsvm, deepsea, reweight</a></li>
<li><a href="http://htmlpreview.github.io/?https://github.com/liangyy/deep_variant/blob/code/pr_roc_curve/report/deepsea_vs_newseq_plot.html">Stage1, train_deepsea, deepsea+extracted, no reweight</a></li>
<li><a href="https://rawgit.com/liangyy/deep_variant/code/pr_roc_curve/report/type1.reweight_plot.html">Stage1, train_deepsea, deepsea+extracted, reweight</a></li>
<li><a href="https://rawgit.com/liangyy/deep_variant/code/pr_roc_curve/report/deepsea-stage1stage2.reweight_plot.html">Stage1 Stage2, all methods, deepsea sequence, reweight</a></li>
<li><a href="https://rawgit.com/liangyy/deep_variant/code/pr_roc_curve/report/extracted-stage1stage2.reweight_plot.html">Stage1 Stage2, all methods, extracted sequence, reweight</a></li>
<li><a href="https://rawgit.com/liangyy/deep_variant/code/pr_roc_curve/template/plot_roc_e081e082.html">Stage1 Stage2, all methods, extracted sequence, reweight, 080917 For Xin</a>,</li>
<li><a href="https://rawgit.com/liangyy/deep_variant/code/pr_roc_curve/report/deepsea-paperNSC.reweight_plot.html">paperNSC, all methods, deepsea sequences, reweighted GC</a>,</li>
<li><a href="https://rawgit.com/liangyy/deep_variant/code/pr_roc_curve/report/extracted-paperNSC.reweight_plot.html">paperNSC, all methods, extracted sequences, reweighted GC</a></li>
<li><a href="https://rawgit.com/liangyy/deep_variant/code/pr_roc_curve/report/deepsea-ATAC053018.reweight_plot.html">ATACseq 05/30/18 from Yifan, DeepSEA sequences</a></li>
<li><a href="https://rawgit.com/liangyy/deep_variant/code/pr_roc_curve/report/extracted-ATAC053018.reweight_plot.html">ATACseq 05/30/18 from Yifan, extracted sequences</a></li>
</ul></li>
<li>AUC scores
<ul>
<li><a href="https://rawgit.com/liangyy/deep_variant/code/performance_comparison/report/deep_brain.auc_comparison.html">Stage1, train_deepsea vs. JASPAR, no reweight</a></li>
<li><a href="https://rawgit.com/liangyy/deep_variant/code/performance_comparison/report/deep_brain_reweight.auc_comparison.html">Stage1, train_deepsea vs. JASPAR, reweight</a></li>
<li><a href="https://rawgit.com/liangyy/deep_variant/code/performance_comparison/report/gkm_vs_type1_on_deepsea_test.auc_comparison.html">Stage1, train_deepsea vs. gkmsvm, reweight</a></li>
<li><a href="https://rawgit.com/liangyy/deep_variant/code/performance_comparison/report/deep_brain-stage1stage2.auc_comparison.html">Stage1 Stage2, compare all methods, reweight</a></li>
<li><a href="https://rawgit.com/liangyy/deep_variant/code/performance_comparison/template/auc_summary_bygroup.html">Stage1 Stage2, compare all methods, reweight, by group</a>,</li>
<li><a href="https://rawgit.com/liangyy/deep_variant/code/performance_comparison/report/deep_brain-paperNSC.auc_comparison.html">paperNSC, all methods, reweighted GC</a></li>
<li><a href="https://rawgit.com/liangyy/deep_variant/code/performance_comparison/report/deep_brain-ATAC053018.auc_comparison.html">ATACseq 05/30/18 from Yifan, all methods, reweighted by GC content</a></li>
</ul></li>
</ol>
</div>
<div id="other-analysis" class="section level1">
<h1><span class="header-section-number">5</span> Other analysis</h1>
<p>There are some other directions and issues we need to explore and tackle. The analysis of them are listed here.</p>
<ol style="list-style-type: decimal">
<li><strong>Motif Analysis</strong>:
<ul>
<li>Examine which filter or a set of filters that contribute most to a particular prediction task. See details <a href="motif_analysis.html">Stage1 here</a>. <strong>ONGOING</strong></li>
</ul></li>
<li><strong>Training Strategy</strong>
<ul>
<li>How to boost the model to specialize on a particular prediction task, say on certain tissue type? (change architecture? how to deal with overfitting?) <strong>TODO</strong></li>
</ul></li>
<li><strong>Performance Evaluation</strong>
<ul>
<li>How to measure the performance on imbalanced data. For instance, how to interpret ROC AUC and PR AUC? See related discussion <a href="auc.html">here</a></li>
<li>Redo DeepSEA evaluation on DNase footprint data (See Allelic Imbalance Analysis below)</li>
</ul></li>
<li><strong>Application</strong>
<ul>
<li>IBD GWAS SNPs. See <a href="SNP_enrichment.html">here</a> Rmarkdown script is at <a href="https://uchicago.box.com/s/zdntptc8udymivtyoapxpu365rpo3b18">here</a></li>
</ul></li>
<li><strong>Data Quality</strong>:
<ul>
<li>GC content.
<ul>
<li><a href="https://rawgit.com/liangyy/deep_variant/code/gc_content/report/deepsea.gc_content.html">Stage1 DeepSEA test set</a>,</li>
<li><a href="https://rawgit.com/liangyy/deep_variant/code/gc_content/report/extracted.gc_content.html">Stage1 Sequences extracted from BED file</a></li>
<li><a href="https://rawgit.com/liangyy/deep_variant/code/gc_content/report/deepsea-ATACseq.gc_content.html">Stage2 Deepsea sequence set</a></li>
<li><a href="https://rawgit.com/liangyy/deep_variant/code/gc_content/report/extracted-ATACseq.gc_content.html">Stage2 Extracted sequence set</a></li>
<li><a href="https://rawgit.com/liangyy/deep_variant/code/gc_content/report/deepsea-paperNSC.gc_content.html">paperNSC DeepSEA sequences</a></li>
<li><a href="https://rawgit.com/liangyy/deep_variant/code/gc_content/report/extracted-paperNSC.gc_content.html">paperNSC Extracted sequences</a></li>
<li><a href="https://rawgit.com/liangyy/deep_variant/code/gc_content/report/deepsea-ATAC053018.gc_content.html">ATACseq 05/30/18 from Yifan, DeepSEA sequence set</a></li>
<li><a href="https://rawgit.com/liangyy/deep_variant/code/gc_content/report/extracted-ATAC053018.gc_content.html">ATACseq 05/30/18 from Yifan, extracted sequence set</a></li>
</ul></li>
<li>Similarity between data sets <a href="https://rawgit.com/liangyy/deep_variant/code/annotation_overlap/report/distance.stage1stage2.html">Stage1 Stage2 with Normal Brain data from DeepSEA</a></li>
</ul></li>
<li><strong>Allelic Imbalance Analysis</strong>:
<ul>
<li>DeepSEA results (Table S3). Presented by data set at <a href="allelic_imbalance_deepsea.html">here</a></li>
<li>Rerun analysis considering both imbalanced variants and balanced variants <a href="allelic_imbalance_rerun.html">here</a></li>
</ul></li>
<li><strong>Detecting Regulartory Grammar</strong>:
<ul>
<li>Critical windows are used to detect important “word” in the sequence and potentially the combination of “word” (“phrase”) in the sequence. Here, we first explore the distribution of critical window along the sequence. <a href="https://rawgit.com/liangyy/deep_variant/code/motif_analysis/report/Type1/newseq.matchgc.E081_sliding_window.signal_block.MaskSize_10.html">E081</a>, <a href="https://rawgit.com/liangyy/deep_variant/code/motif_analysis/report/Type1/newseq.matchgc.Noonan_sliding_window.signal_block.MaskSize_10.html">Noonan</a></li>
</ul></li>
</ol>
</div>
<div id="todo-list" class="section level1">
<h1><span class="header-section-number">6</span> TODO list</h1>
<ul>
<li>4/9/17:
<ol style="list-style-type: decimal">
<li><del>ROC/PR AUC interpretation</del> <a href="auc.html">link</a></li>
<li><del>Extract sequences from new data sets</del> <a href="memo_extract.html">link</a></li>
</ol></li>
<li>4/10/17:
<ol start="3" style="list-style-type: decimal">
<li>New loss function? Weighted hinge loss? <a href="loss_function.html">link</a> <strong>TODO</strong></li>
<li>Motif analysis of current model <a href="motif_analysis.html">link</a> <strong>ONGOING</strong></li>
</ol></li>
<li>4/14/17:
<ol start="5" style="list-style-type: decimal">
<li><del>Build up pipeline using snakemake.</del> See issue1 at <a href="https://github.com/CreRecombinase/DeepVariantPrediction">link</a></li>
</ol></li>
<li>4/26/17:
<ol start="6" style="list-style-type: decimal">
<li><del>Build baseline classifier</del> <a href="baseline.html">link</a></li>
</ol></li>
<li>5/29/17:
<ol start="7" style="list-style-type: decimal">
<li>GC content bias in training and testing data
<ul>
<li>To overcome this, we can either use unbiased sequence set or reweight the performance evaluation function based on the GC content.</li>
<li>At first, do reweighting (<del>testing</del> - training - <del>motif analysis</del>) <strong>ONGOING</strong> see <a href="https://rawgit.com/liangyy/deep_variant/code/performance_comparison/report/deep_brain_reweight.auc_comparison.html">testing with reweight</a> and <a href="motif_analysis.html">motif analysis with gc matched sequences</a></li>
<li>Construct a sequence set with matched GC content in positive and negative instances (use Homer?) <strong>TODO</strong></li>
</ul></li>
</ol></li>
<li>5/31/17:
<ol start="8" style="list-style-type: decimal">
<li>Search for motif hits using top 20 motifs in motif activation analysis by GC matched sequences <strong>TODO</strong></li>
</ol></li>
</ul>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
