<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Yanyu Liang" />


<title>Memo - annotate existing sequences with new labels (ATAC-seq from Noonan’s lab)</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-1.1/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-1.1/highlight.js"></script>
<link href="site_libs/font-awesome-4.5.0/css/font-awesome.min.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}

.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Deep Variant</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Tasks</a>
</li>
<li>
  <a href="analysis_list.html">Analysis</a>
</li>
<li>
  <a href="memo_list.html">Technical Memo</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/liangyy/deep_variant">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Memo - annotate existing sequences with new labels (ATAC-seq from Noonan’s lab)</h1>
<h4 class="author"><em>Yanyu Liang</em></h4>
<h4 class="date"><em>29 June, 2017</em></h4>

</div>


<style>
pre code, pre, code {
  white-space: pre !important;
  overflow-x: scroll !important;
  word-break: keep-all !important;
  word-wrap: initial !important;
}
</style>
<div id="data" class="section level1">
<h1><span class="header-section-number">1</span> Data</h1>
<pre class="bash"><code>$ /project2/xinhe/yanyul/deep_variant/yanyu/deep_brain/data
$ cp ../ASD/other_annotation/epigenomic_annotation/Noonan_hNSC_*bb deep_variant/yanyu/deep_brain/data/</code></pre>
</div>
<div id="bigbed-to-bed" class="section level1">
<h1><span class="header-section-number">2</span> BigBed to Bed</h1>
<pre class="bash"><code>$ bash process_ATACseq_from_Noonan.sh</code></pre>
</div>
<div id="label-deepsea-sequence-with-atac-seq-signal" class="section level1">
<h1><span class="header-section-number">3</span> Label DeepSEA sequence with ATAC-seq signal</h1>
<pre class="bash"><code>$ cd ../
$ python my_scripts/wrapper_label_intervals.py data/allTFs.pos.withID.bed data/0628_ATACseq_list.txt test
awk: cmd. line:1: (FILENAME=- FNR=4097) fatal: print to &quot;standard output&quot; failed (Broken pipe)
cat: write error: Broken pipe
mkdir test/0628_ATACseq_list.txt
&gt;&gt;&gt; working on data/Noonan_hNSC_50_peaks.bed
&gt;&gt;&gt; &gt;&gt;&gt; sort data/Noonan_hNSC_50_peaks.bed
&gt;&gt;&gt; &gt;&gt;&gt; checking data/Noonan_hNSC_50_peaks.bed
&gt;&gt;&gt; &gt;&gt;&gt; number of columns data/Noonan_hNSC_50_peaks.bed
awk: cmd. line:1: (FILENAME=- FNR=2731) fatal: print to &quot;standard output&quot; failed (Broken pipe)
cat: write error: Broken pipe
&gt;&gt;&gt; &gt;&gt;&gt; intersecting data/Noonan_hNSC_50_peaks.bed
&gt;&gt;&gt; previous = 124826, after = 124826
&gt;&gt;&gt; working on data/Noonan_hNSC_P15-1_peaks.bed
&gt;&gt;&gt; &gt;&gt;&gt; sort data/Noonan_hNSC_P15-1_peaks.bed
&gt;&gt;&gt; &gt;&gt;&gt; checking data/Noonan_hNSC_P15-1_peaks.bed
&gt;&gt;&gt; &gt;&gt;&gt; number of columns data/Noonan_hNSC_P15-1_peaks.bed
awk: cmd. line:1: (FILENAME=- FNR=2731) fatal: print to &quot;standard output&quot; failed (Broken pipe)
cat: write error: Broken pipe
&gt;&gt;&gt; &gt;&gt;&gt; intersecting data/Noonan_hNSC_P15-1_peaks.bed
&gt;&gt;&gt; previous = 84105, after = 84105
&gt;&gt;&gt; working on data/Noonan_hNSC_P5-1_peaks.bed
&gt;&gt;&gt; &gt;&gt;&gt; sort data/Noonan_hNSC_P5-1_peaks.bed
&gt;&gt;&gt; &gt;&gt;&gt; checking data/Noonan_hNSC_P5-1_peaks.bed
&gt;&gt;&gt; &gt;&gt;&gt; number of columns data/Noonan_hNSC_P5-1_peaks.bed
awk: cmd. line:1: (FILENAME=- FNR=2731) fatal: print to &quot;standard output&quot; failed (Broken pipe)
cat: write error: Broken pipe
&gt;&gt;&gt; &gt;&gt;&gt; intersecting data/Noonan_hNSC_P5-1_peaks.bed
&gt;&gt;&gt; previous = 67160, after = 67160
&gt;&gt;&gt; working on data/Noonan_hNSC_P5-2_peaks.bed
&gt;&gt;&gt; &gt;&gt;&gt; sort data/Noonan_hNSC_P5-2_peaks.bed
&gt;&gt;&gt; &gt;&gt;&gt; checking data/Noonan_hNSC_P5-2_peaks.bed
&gt;&gt;&gt; &gt;&gt;&gt; number of columns data/Noonan_hNSC_P5-2_peaks.bed
awk: cmd. line:1: (FILENAME=- FNR=2731) fatal: print to &quot;standard output&quot; failed (Broken pipe)
cat: write error: Broken pipe
&gt;&gt;&gt; &gt;&gt;&gt; intersecting data/Noonan_hNSC_P5-2_peaks.bed
&gt;&gt;&gt; previous = 86563, after = 86563</code></pre>
</div>
<div id="merge-labels-and-prepare-input-file-for-training" class="section level1">
<h1><span class="header-section-number">4</span> Merge labels and prepare input file for training</h1>
<div id="merge-all-labels" class="section level2">
<h2><span class="header-section-number">4.1</span> Merge all labels</h2>
<pre class="bash"><code>$ cd /project2/xinhe/yanyul/deep_variant/yanyu/deep_brain
$ cd data
$ ls Noonan_h*intersect &gt; ../test/0629_ATACseq_merged/merge.txt
$ python my_scripts/merge_labels.py test/0628_ATACseq_list.txt/label.hdf5 test/0629_ATACseq_merged/merge.txt test/0629_ATACseq_merged/</code></pre>
</div>
<div id="combine-labels-with-sequences" class="section level2">
<h2><span class="header-section-number">4.2</span> Combine labels with sequences</h2>
<div id="training" class="section level3">
<h3><span class="header-section-number">4.2.1</span> Training</h3>
<pre class="bash"><code>$ python my_scripts/add_y_label_and_rename.py my_train/DeepSEA_train_12__with_label.hdf5 test/0629_ATACseq_merged/label_merge.hdf5
HDF5 &quot;my_train/DeepSEA_train_12__with_label.hdf5&quot; {
GROUP &quot;/&quot; {
   DATASET &quot;traindata&quot; {
      DATATYPE  H5T_IEEE_F64LE
      DATASPACE  SIMPLE { ( 4400000, 4 ) / ( 4400000, 4 ) }
   }
   DATASET &quot;trainxdata&quot; {
      DATATYPE  H5T_IEEE_F32LE
      DATASPACE  SIMPLE { ( 4400000, 925 ) / ( 4400000, 925 ) }
   }
}
}
HDF5 &quot;test/0629_ATACseq_merged/label_merge.hdf5&quot; {
GROUP &quot;/&quot; {
   DATASET &quot;merged&quot; {
      DATATYPE  H5T_IEEE_F64LE
      DATASPACE  SIMPLE { ( 2608182, 4 ) / ( 2608182, 4 ) }
   }
}
}
$ python my_scripts/add_y_label_and_rename.py my_train/DeepSEA_train_12__with_label.hdf5 test/0629_ATACseq_merged/label_merge.hdf5 trainxdata merged 1 2200000 test/0629_ATACseq_merged/</code></pre>
</div>
<div id="testing" class="section level3">
<h3><span class="header-section-number">4.2.2</span> Testing</h3>
<pre class="bash"><code>$ python my_scripts/add_y_label_and_rename.py my_train/DeepSEA_test_12__with_label.hdf5 test/0629_ATACseq_merged/label_merge.hdf5
HDF5 &quot;my_train/DeepSEA_test_12__with_label.hdf5&quot; {
GROUP &quot;/&quot; {
   DATASET &quot;traindata&quot; {
      DATATYPE  H5T_IEEE_F64LE
      DATASPACE  SIMPLE { ( 455024, 4 ) / ( 455024, 4 ) }
   }
   DATASET &quot;trainxdata&quot; {
      DATATYPE  H5T_IEEE_F32LE
      DATASPACE  SIMPLE { ( 455024, 925 ) / ( 455024, 925 ) }
   }
}
}
HDF5 &quot;test/0629_ATACseq_merged/label_merge.hdf5&quot; {
GROUP &quot;/&quot; {
   DATASET &quot;merged&quot; {
      DATATYPE  H5T_IEEE_F64LE
      DATASPACE  SIMPLE { ( 2608182, 4 ) / ( 2608182, 4 ) }
   }
}
}
$ python my_scripts/add_y_label_and_rename.py my_train/DeepSEA_test_12__with_label.hdf5 test/0629_ATACseq_merged/label_merge.hdf5 trainxdata merged 2309367 2536878 test/0629_ATACseq_merged/</code></pre>
</div>
<div id="validation" class="section level3">
<h3><span class="header-section-number">4.2.3</span> Validation</h3>
<pre class="bash"><code>$ python my_scripts/add_y_label_and_rename.py my_train/DeepSEA_valid_12__with_label.hdf5 test/0629_ATACseq_merged/label_merge.hdf5
HDF5 &quot;my_train/DeepSEA_valid_12__with_label.hdf5&quot; {
GROUP &quot;/&quot; {
   DATASET &quot;traindata&quot; {
      DATATYPE  H5T_IEEE_F64LE
      DATASPACE  SIMPLE { ( 8000, 4 ) / ( 8000, 4 ) }
   }
   DATASET &quot;trainxdata&quot; {
      DATATYPE  H5T_IEEE_F32LE
      DATASPACE  SIMPLE { ( 8000, 925 ) / ( 8000, 925 ) }
   }
}
}
HDF5 &quot;test/0629_ATACseq_merged/label_merge.hdf5&quot; {
GROUP &quot;/&quot; {
   DATASET &quot;merged&quot; {
      DATATYPE  H5T_IEEE_F64LE
      DATASPACE  SIMPLE { ( 2608182, 4 ) / ( 2608182, 4 ) }
   }
}
}
$ python my_scripts/add_y_label_and_rename.py my_train/DeepSEA_valid_12__with_label.hdf5 test/0629_ATACseq_merged/label_merge.hdf5 trainxdata merged 2200001 2204000 test/0629_ATACseq_merged/</code></pre>
</div>
<div id="summary" class="section level3">
<h3><span class="header-section-number">4.2.4</span> Summary</h3>
<pre class="bash"><code>$ h5dump -A -H test/0629_ATACseq_merged/*with_label.hdf5
HDF5 &quot;test/0629_ATACseq_merged/DeepSEA_test_12__with_label_with_label.hdf5&quot; {
GROUP &quot;/&quot; {
   DATASET &quot;traindata&quot; {
      DATATYPE  H5T_IEEE_F64LE
      DATASPACE  SIMPLE { ( 455024, 4 ) / ( 455024, 4 ) }
   }
   DATASET &quot;trainxdata&quot; {
      DATATYPE  H5T_IEEE_F32LE
      DATASPACE  SIMPLE { ( 455024, 925 ) / ( 455024, 925 ) }
   }
}
}
HDF5 &quot;test/0629_ATACseq_merged/DeepSEA_train_12__with_label_with_label.hdf5&quot; {
GROUP &quot;/&quot; {
   DATASET &quot;traindata&quot; {
      DATATYPE  H5T_IEEE_F64LE
      DATASPACE  SIMPLE { ( 4400000, 4 ) / ( 4400000, 4 ) }
   }
   DATASET &quot;trainxdata&quot; {
      DATATYPE  H5T_IEEE_F32LE
      DATASPACE  SIMPLE { ( 4400000, 925 ) / ( 4400000, 925 ) }
   }
}
}
HDF5 &quot;test/0629_ATACseq_merged/DeepSEA_valid_12__with_label_with_label.hdf5&quot; {
GROUP &quot;/&quot; {
   DATASET &quot;traindata&quot; {
      DATATYPE  H5T_IEEE_F64LE
      DATASPACE  SIMPLE { ( 8000, 4 ) / ( 8000, 4 ) }
   }
   DATASET &quot;trainxdata&quot; {
      DATATYPE  H5T_IEEE_F32LE
      DATASPACE  SIMPLE { ( 8000, 925 ) / ( 8000, 925 ) }
   }
}
}</code></pre>
</div>
</div>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
