<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />




<title>Deep brain - Type 1 - evaluation on allelic imbalance instances</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-1.1/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-1.1/highlight.js"></script>
<link href="site_libs/font-awesome-4.5.0/css/font-awesome.min.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}

.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Deep Variant</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Tasks</a>
</li>
<li>
  <a href="analysis_list.html">Analysis</a>
</li>
<li>
  <a href="memo_list.html">Technical Memo</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/liangyy/deep_variant">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Deep brain - Type 1 - evaluation on allelic imbalance instances</h1>
<h4 class="date"><em>11 May, 2017</em></h4>

</div>


<style>
pre code, pre, code {
        white-space: pre !important;
        overflow-x: scroll !important;
        word-break: keep-all !important;
        word-wrap: initial !important;
    }
</style>
<div id="about-alignment-bias" class="section level1">
<h1><span class="header-section-number">1</span> About alignment bias</h1>
<p>It has not been implemented. <strong>TODO</strong></p>
<p>The basic steps to deal with alignment bias is as follow:</p>
<ol style="list-style-type: decimal">
<li>Align and call all possible heterozygous site</li>
<li>Use <code>WASP</code> to look for reads at possible heterozygous site</li>
<li>Re-align those reads and filter out the ones that cannot be aligned to the same position after flip the base</li>
<li>Call heterozygous site with quality control and compute the degree of imbalance</li>
</ol>
</div>
<div id="results" class="section level1">
<h1><span class="header-section-number">2</span> Results</h1>
<div id="load-data" class="section level2">
<h2><span class="header-section-number">2.1</span> Load Data</h2>
<pre class="r"><code>source(&#39;my_r.R&#39;)
library(feather)
library(ggplot2)
library(reshape2)
variant_url &lt;- &#39;https://github.com/liangyy/deep_variant/blob/code/allelic_imbalance/data/E081.combined.imbalance.gz?raw=true&#39;
variant &lt;- read.table(read_gz_url(variant_url), sep = &#39;\t&#39;, header = T)
score_url &lt;- &#39;https://raw.githubusercontent.com/CreRecombinase/DeepVariantPrediction/issue1/score/keras_deepsea_with_040417_2_42_head_copy/allelic_imbalance_E081.E081_allelic_imbalance_result.feather&#39;
score &lt;- read_feather_url(score_url)</code></pre>
<pre><code>## Warning: Coercing int64 to double</code></pre>
<pre class="r"><code>variant$Ref &lt;- NA
variant$Ref[score$Varient.ID] &lt;- score$Allele1
variant$Alt &lt;- NA
variant$Alt[score$Varient.ID] &lt;- score$Allele2</code></pre>
</div>
<div id="call-imbalance" class="section level2">
<h2><span class="header-section-number">2.2</span> Call Imbalance</h2>
<pre class="r"><code>imbalance.threshold &lt;- 0.01 # Fisher&#39;s exact test p-value
variant$Imbalance &lt;- &#39;Balanced&#39;
variant$Imbalance[variant$P.Value &lt; imbalance.threshold] &lt;- &#39;Imbalanced&#39;</code></pre>
</div>
<div id="quality-control" class="section level2">
<h2><span class="header-section-number">2.3</span> Quality Control</h2>
<div id="remove-strand-imbalanced-instances" class="section level3">
<h3><span class="header-section-number">2.3.1</span> Remove strand imbalanced instances</h3>
<p><span class="math inline">\(\frac{|read_+ - read_-|}{read_+ + read_-} &lt; 0.3\)</span></p>
<pre class="r"><code>strand.threshold &lt;- 0.3
strand.read1 &lt;- abs(variant$Reads1Plus - variant$Reads1Minus) / (variant$Reads1Plus + variant$Reads1Minus)
strand.read2 &lt;- abs(variant$Reads2Plus - variant$Reads2Minus) / (variant$Reads2Plus + variant$Reads2Minus)
ggplot(as.data.frame(cbind(strand.read1, strand.read2))) + geom_bin2d(aes(x = strand.read1, y = strand.read2)) + ggtitle(&#39;Score of Strand Imbalance&#39;)</code></pre>
<p><img src="type1_allelic_imbalance_files/figure-html/unnamed-chunk-3-1.png" width="672" /></p>
<pre class="r"><code>strand.pass &lt;- (strand.read1 &lt; strand.threshold) &amp; (strand.read2 &lt; strand.threshold)</code></pre>
<p>The asymmetric result of read1 and read2 may come from alignment bias.</p>
</div>
<div id="remove-low-coverage-instances" class="section level3">
<h3><span class="header-section-number">2.3.2</span> Remove low coverage instances</h3>
<pre class="r"><code>coverage.threshold &lt;- 100
coverage &lt;- variant$Reads1 + variant$Reads2
coverage.pass &lt;- coverage &gt; coverage.threshold</code></pre>
</div>
</div>
<div id="qc-summary" class="section level2">
<h2><span class="header-section-number">2.4</span> QC Summary</h2>
<pre class="r"><code>both.pass &lt;- strand.pass &amp; coverage.pass
variant$pass &lt;- both.pass
temp &lt;- as.data.frame(cbind(strand.pass, coverage.pass, both.pass))
temp &lt;- melt(temp, id.vars = c())
temp &lt;- temp[temp$value,]
ggplot(temp) + geom_bar(aes(x = variable)) + ggtitle(&#39;Quality Control \n Count of bad instances&#39;)</code></pre>
<p><img src="type1_allelic_imbalance_files/figure-html/unnamed-chunk-5-1.png" width="672" /></p>
<pre class="r"><code>ggplot(variant) + geom_bar(aes(x = both.pass, fill = Imbalance), position = &#39;dodge&#39;) + ggtitle(&#39;Filtering Result&#39;)</code></pre>
<p><img src="type1_allelic_imbalance_files/figure-html/unnamed-chunk-5-2.png" width="672" /></p>
</div>
<div id="predicted-imbalance" class="section level2">
<h2><span class="header-section-number">2.5</span> Predicted imbalance</h2>
<div id="overview" class="section level3">
<h3><span class="header-section-number">2.5.1</span> Overview</h3>
<pre class="r"><code>variant.pass &lt;- variant[variant$pass,]
ggplot(variant.pass) + geom_point(aes(x = Alt, y = Ref, color = log10(Odds.Ratio))) + scale_colour_gradient2() + facet_grid(.~Imbalance) + geom_abline(intercept = 0.07, slope = 1) + geom_abline(intercept = -0.07, slope = 1) + ggtitle(&#39;The Predicted Effect of \n Heterozygous sites in DNase-seq of E081&#39;)</code></pre>
<p><img src="type1_allelic_imbalance_files/figure-html/unnamed-chunk-6-1.png" width="672" /></p>
</div>
<div id="observed-log-odds-ratio-vs.predicted-log-odds-ratio" class="section level3">
<h3><span class="header-section-number">2.5.2</span> Observed Log Odds Ratio vs. Predicted Log Odds Ratio</h3>
<pre class="r"><code>variant.pass$Log10.Odds.Ratio.Observed &lt;- log10(variant.pass$Odds.Ratio)
variant.pass$Log10.Odds.Ratio.Predicted &lt;- log10(variant.pass$Ref / variant.pass$Alt)
model.linear = lm(Log10.Odds.Ratio.Observed ~ 0 + Log10.Odds.Ratio.Predicted, data = variant.pass)
ggplot(variant.pass, aes(x = Log10.Odds.Ratio.Predicted, y = Log10.Odds.Ratio.Observed)) + geom_point(aes(color = Imbalance)) + geom_smooth(method=&#39;lm&#39;,formula=y~0+x) + annotate(&quot;text&quot;, x = 0.4, y = -1, label = lm_eqn(model.linear), parse = TRUE) + ggtitle(&#39;Observed Log Odds Ratio vs \n Predicted Log Odds Ratio&#39;)</code></pre>
<p><img src="type1_allelic_imbalance_files/figure-html/unnamed-chunk-7-1.png" width="672" /></p>
</div>
<div id="predictive-power-of-predicted-pref-and-palt" class="section level3">
<h3><span class="header-section-number">2.5.3</span> Predictive power of predicted p(ref) and p(alt)</h3>
<p>Model: imbalance ~ f(ref, alt), link = binomial. <span class="math inline">\(f_1 = |x - y|\)</span>. <span class="math inline">\(f_2 = |\log_{10}(x/(1-x)) - \log_{10}(y/(1-y))|\)</span></p>
<pre class="r"><code>model.logistic.abs &lt;- glm(as.numeric(Imbalance == &#39;Imbalanced&#39;) ~ abs(Ref - Alt), family=binomial(link=&#39;logit&#39;), data=variant.pass)
model.logistic.lor &lt;- glm(as.numeric(Imbalance == &#39;Imbalanced&#39;) ~ abs(Log10.Odds.Ratio.Predicted), family=binomial(link=&#39;logit&#39;), data=variant.pass)</code></pre>
<pre class="r"><code>library(pander)
panderOptions(&#39;knitr.auto.asis&#39;, FALSE)
pander(model.logistic.abs)</code></pre>
<table style="width:90%;">
<caption>Fitting generalized (binomial/logit) linear model: as.numeric(Imbalance == “Imbalanced”) ~ abs(Ref - Alt)</caption>
<colgroup>
<col width="29%" />
<col width="15%" />
<col width="18%" />
<col width="13%" />
<col width="13%" />
</colgroup>
<thead>
<tr class="header">
<th align="center"> </th>
<th align="center">Estimate</th>
<th align="center">Std. Error</th>
<th align="center">z value</th>
<th align="center">Pr(&gt;|z|)</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="center"><strong>abs(Ref - Alt)</strong></td>
<td align="center">4.601</td>
<td align="center">1.137</td>
<td align="center">4.045</td>
<td align="center">5.239e-05</td>
</tr>
<tr class="even">
<td align="center"><strong>(Intercept)</strong></td>
<td align="center">-0.7945</td>
<td align="center">0.04488</td>
<td align="center">-17.7</td>
<td align="center">4.024e-70</td>
</tr>
</tbody>
</table>
<pre class="r"><code>pander(model.logistic.lor)</code></pre>
<table>
<caption>Fitting generalized (binomial/logit) linear model: as.numeric(Imbalance == “Imbalanced”) ~ abs(Log10.Odds.Ratio.Predicted)</caption>
<colgroup>
<col width="46%" />
<col width="13%" />
<col width="15%" />
<col width="12%" />
<col width="12%" />
</colgroup>
<thead>
<tr class="header">
<th align="center"> </th>
<th align="center">Estimate</th>
<th align="center">Std. Error</th>
<th align="center">z value</th>
<th align="center">Pr(&gt;|z|)</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="center"><strong>abs(Log10.Odds.Ratio.Predicted)</strong></td>
<td align="center">5.186</td>
<td align="center">0.6739</td>
<td align="center">7.696</td>
<td align="center">1.405e-14</td>
</tr>
<tr class="even">
<td align="center"><strong>(Intercept)</strong></td>
<td align="center">-0.899</td>
<td align="center">0.04614</td>
<td align="center">-19.49</td>
<td align="center">1.458e-84</td>
</tr>
</tbody>
</table>
<pre class="r"><code>library(precrec)
joined &lt;- join_scores(model.logistic.abs$fitted.values, model.logistic.lor$fitted.values)
msmdat &lt;- mmdata(joined, as.numeric(variant.pass$Imbalance == &#39;Imbalanced&#39;), modnames = c(&#39;f1&#39;, &#39;f2&#39;))
mscurves &lt;- evalmod(msmdat)
plot(mscurves)</code></pre>
<p><img src="type1_allelic_imbalance_files/figure-html/unnamed-chunk-10-1.png" width="672" /></p>
<pre class="r"><code>pander(auc(mscurves))</code></pre>
<table style="width:53%;">
<colgroup>
<col width="15%" />
<col width="11%" />
<col width="18%" />
<col width="8%" />
</colgroup>
<thead>
<tr class="header">
<th align="center">modnames</th>
<th align="center">dsids</th>
<th align="center">curvetypes</th>
<th align="center">aucs</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="center">f1</td>
<td align="center">1</td>
<td align="center">ROC</td>
<td align="center">0.5168</td>
</tr>
<tr class="even">
<td align="center">f1</td>
<td align="center">1</td>
<td align="center">PRC</td>
<td align="center">0.3674</td>
</tr>
<tr class="odd">
<td align="center">f2</td>
<td align="center">1</td>
<td align="center">ROC</td>
<td align="center">0.5722</td>
</tr>
<tr class="even">
<td align="center">f2</td>
<td align="center">1</td>
<td align="center">PRC</td>
<td align="center">0.417</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
